eng/build.yml

```diff
@@ -187,3 +187,107 @@ jobs:
         publishLocation: Container	
       continueOnError: true	
       condition: always()
+
+- ${{ if in(parameters.agentOs, 'Windows_NT', 'Darwin') }}:
```

Comment:
We have to run the AOT tests in a separate `job` (however they can still be in the same pipeline), due to https://github.com/dotnet/sdk/issues/22655. We may be able to consolidate the AOT tests into the existing test job once that issue is resolved.